### PR TITLE
Image refresh for arch

### PR DESCRIPTION
**Pull request by cockpituous**
Image refresh for arch
 * [x] image-refresh arch

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20240720-224801

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-arch-5b58ef51-20240720-223718/log.html

**Comment by tomasmatus** [comment]
```
log: osinfo-detect command failed:  (process:5193): GLib-GIO-WARNING **: 08:01:10.590: Can't find module 'gvfs' specified in GIO_USE_VFS
```
note to self tomorrow - installing `gvfs` package seems to have fixed the failing tests, tested locally (investigate what's up?)

**Comment by martinpitt** [comment]
@tomasmatus Hmm, according to the Removed/Added/Changed, gvfs didn't drop and libosinfo didn't update. gsettings-desktop-schemas was removed though -- did installing gvfs pull in some of the removed packages as a dependency? gvfs is rather heavy and a desktop package, it should preferably not be on a server image.

**Comment by martinpitt** [comment]
@tomasmatus also, please check that you *don't* get that GVFS warning in the previous image and others, i.e. that it is *actually* a symptom of the regression and not just unrelated. 

For example, we've been getting that warning for many years e.g. [here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1730-e2fdd3c4-20240719-133438-rhel-9-4/log)

So my bet is that installing gvfs pulled in some other package which is an undeclared dependency of libosinfo. That should be reported to/fixed in arch, and then we can install it manually as a workaround.

Thanks!

**Comment by tomasmatus** [comment]
```
+ ./image-diff /cache/images/arch-80f19ce9b6c04a1438efe4ed4eeb821f9b90139e8cc18e8cad2c90be1733e0e5.qcow2 arch
Removed:
  adobe-source-code-pro-fonts (2.042u+1.062i+1.026vf-2)
  cantarell-fonts (1:0.303.1-1)
  dconf (0.40.0-2)
  gsettings-desktop-schemas (46.0-2)
  ```
  both `dconf` and `gsettings-desktop-schemas` got pulled by the [`gvfs`](https://archlinux.org/packages/extra/x86_64/gvfs/) package. Installing [`gsettings-desktop-schema`](https://archlinux.org/packages/extra/any/gsettings-desktop-schemas/) also pulls `dconf` so it looks like both of these packages are needed.

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20240723-085404

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-arch-a0adcf81-20240723-084324/log.html

File: images/scripts/arch.setup [hunk 1]
```diff
@@ -12,6 +12,7 @@ export LC_ALL=C
 COCKPIT_DEPS="\
 binutils \
 edk2-ovmf \
+gsettings-desktop-schemas \
```
Comment (martinpitt) [inline images/scripts/arch.setup:15]: This is highly dubious. This is totally *not* a cockpit dependency. Please report a bug against arch about this (libosinfo has a CLI which should reproduce this). libosinfo should need a desktop package, and if it really really has to, it needs to declare a dependency by itself. After that, please move this to TEST_PACKAGES with a `# HACK` comment. Thanks!

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20240723-125425

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-arch-f7c6989c-20240723-124311/log.html

**Comment by tomasmatus** [comment]
libosinfo on arch should declare `gsettings-desktop-schemas` as a dependency

```
(process:7692): GLib-GIO-ERROR **: 08:02:38.623: Settings schema 'org.gnome.system.proxy' is not installed
log: osinfo-detect command failed:  (process:7744): GLib-GIO-WARNING **: 08:02:43.630: Can't find module 'gvfs' specified in GIO_USE_VFS
```
[full log here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-6643-02ccced1-20240721-075151-arch-cockpit-project-cockpit-machines/log.html#13)

**Comment by martinpitt** [comment]
@tomasmatus does that reproduce with calling `osinfo-detect` on the CLI? I.e. is that a reproducer that can be reported to arch?

**Comment by tomasmatus** [comment]
@martinpitt here's the reproducer

```
[root@archlinux ~]# osinfo-detect 'https://archive.fedoraproject.org/pub/archive/fedora/linux/releases/28/Server/x86_64/os'

(osinfo-detect:1759): GLib-GIO-ERROR **: 07:29:03.513: Settings schema 'org.gnome.system.proxy' is not installed
Trace/breakpoint trap (core dumped)
```

**Review by martinpitt — CHANGES_REQUESTED** [review]
Cheers, nice work! Please squash the two image refresh commits together, force-push, and trigger the tests.

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**